### ISSUE TITLE
Error "omnifunction: jedi#completions moves cursor!"

### ISSUE THREAD

**Issue by blueyed** [Shougo/deoplete.nvim#75]
I am seeing this error with jedi-vim omni completions when typing too fast.

> [deoplete] omnifunction: jedi#completions moves cursor!
> '"gather_results"'
> [deoplete] Deoplete cannot support it in omni source. You should use g:deoplete#omni_patterns.

TEST CASE:
1. type `import os` in a Python buffer, without waiting long enough after the space

It can be triggered by typing "import os", where the difference is if you wait
after the space for the initial completions or not.

I've added some `winsaveview`/`winrestview` calls in the `jedi#completions` (https://github.com/davidhalter/jedi-vim/blob/master/autoload/jedi.vim#L239-L241) function, but that still gave the error.

`:mess` from when the error does not happen:
'"gather_results"'
'"source: omni, input: "'
…
'"gather_results"'
'"source: omni, input: i"'
…
'"gather_results"'
'"source: omni, input: im"'
…
'"gather_results"'
'"source: omni, input: import "'
…

bad:
'"gather_results"'
'"source: omni, input: "'
…
'"gather_results"'
'"source: omni, input: i"'
…
'"gather_results"'
'"source: omni, input: im"'
…
'"gather_results"'
'"source: omni, input: import "'
…
[deoplete] omnifunction: jedi#completions moves cursor!
'"gather_results"'
[deoplete] Deoplete cannot support it in omni source. You should use g:deoplete#omni_patterns.
'"source: omni, input: import os"'
…

**Comment by blueyed** [comment]
It happens because the position gets changed by the user: https://github.com/Shougo/deoplete.nvim/blob/master/rplugin/python3/deoplete/sources/omni.py#L75-L85

Maybe this check should not be there altogether?
Or bound / updated through `CursorMovedI`?

And then if the user continued typing inbetween, then the outdated completions should be ignored / thrown away at this place?

**Comment by Shougo** [comment]
@bfredl Did you read the FAQ?   I think you don't disable `g:jedi#smart_auto_mappings`.

```vim
	autocmd FileType python setlocal omnifunc=jedi#completions
	let g:jedi#completions_enabled = 0
	let g:jedi#auto_vim_configuration = 0
	let g:jedi#smart_auto_mappings = 0
	let g:jedi#show_call_signatures = 0
```

**Comment by Shougo** [comment]
Of course, you can create jedi source for deoplete.
It is better.

**Comment by Shougo** [comment]
> I've added some winsaveview/winrestview calls in the jedi#completions (https://github.com/davidhalter/jedi-vim/blob/master/autoload/jedi.vim#L239-L241) function, but that still gave the error.

It is no means.  Because, deoplete is asynchronous.

**Comment by Shougo** [comment]
I cannot reproduce the problem.
But I have improved the cursor check.

**Comment by blueyed** [comment]
I was using `let g:jedi#smart_auto_mappings = 0`.

Your fix (3642362462eedba1c3c7beaf108783b1fab74d76) helped.
Thanks!
